### PR TITLE
[enhancement](compaction) add idle schedule and max_size limit for base compaction

### PR DESCRIPTION
**Pull request by yixiutt**
1.add idle schedule for base compaction
2.dup key don't do base if size exceed & no delete predict

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/base_compaction.cpp [hunk 1]
```diff
@@ -81,16 +84,61 @@ Status BaseCompaction::execute_compact_impl() {
     return Status::OK();
 }
 
+bool BaseCompaction::_filter_input_rowset() {
+    bool exceed = false;
+    // if enable dup key skip big file and no delete predicate
+    // we skip big files too save resources
+    if (!config::enable_dup_key_base_compaction_skip_big_file ||
+        _tablet->keys_type() != KeysType::DUP_KEYS || _tablet->delete_predicates().size() != 0) {
+        return false;
```
Comment (dataroaring) [inline be/src/olap/base_compaction.cpp:93]: Could we check version of delete predicates? e.g. If  we dont have delete predicates before version 10, then we can skip large rowset before version 10.

File: be/src/olap/base_compaction.cpp [hunk 2]
```diff
@@ -81,16 +84,61 @@ Status BaseCompaction::execute_compact_impl() {
...
+        }
+        if (total_size > max_size) {
+            exceed = true;
+            break;
+        }
+    }
+    // remove rowset left
+    while (rs_iter != _input_rowsets.end()) {
+        rs_iter = _input_rowsets.erase(rs_iter);
+    }
```
Comment (dataroaring) [inline be/src/olap/base_compaction.cpp:124]: Here, we limit total size of inputs to base_compaction_dup_key_max_file_size_mbytes? e.g. if there are four rowsets with 512MB size, we just compact the first 2?

**Review by dataroaring — CHANGES_REQUESTED** [review]

File: be/src/olap/base_compaction.cpp (hunk 2 above)
Comment (yixiutt) [reply be/src/olap/base_compaction.cpp]: fixed, we will compact four 512M rowset

File: be/src/olap/base_compaction.cpp [hunk 3]
```diff
@@ -81,16 +84,35 @@ Status BaseCompaction::execute_compact_impl() {
...
 }
 
+void BaseCompaction::_filter_input_rowset() {
+    // if enable dup key skip big file and no delete predicate
+    // we skip big files too save resources
+    if (!config::enable_dup_key_base_compaction_skip_big_file ||
+        _tablet->keys_type() != KeysType::DUP_KEYS || _tablet->delete_predicates().size() != 0) {
+        return;
+    }
+    int64_t max_size = config::base_compaction_dup_key_max_file_size_mbytes;
```
Comment (zhannngchen) [inline be/src/olap/base_compaction.cpp:94]: should be `base_compaction_dup_key_max_file_size_mbytes * 1024 * 1024`?

Comment (yixiutt) [reply be/src/olap/base_compaction.cpp]: fixed

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**